.github/workflows/build.yml

```diff
@@ -19,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [windows-2019, ubuntu-latest, macos-14]
+        os: [windows-2019, ubuntu-latest, macos-12]
```

Comment:
Why the downgrade to 12?

Reply:
Since the removal of macOS-11, we haven't had a working verification build for macOS, except on the Eclipse CI. See https://github.com/eclipse-theia/theia-blueprint/actions/workflows/build.yml . I started addressing this issue by aligning our approach with that of the CDT Cloud Blueprint, as seen here: https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/commit/fddcf9a725876637710e8d068730f11484497b12

Generally, I believe it's better to use older machines for verification jobs because if something works on macOS 12, it's likely to work on macOS 14 as well. However, if it only runs on macOS 14, we cannot be sure it will work on macOS 12.